sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -74,12 +76,37 @@ class QueryExecution(
     sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
   }
 
+  // SPARK-35378: Eagerly execute Command
```

Comment:
```
SPARK-35378: Commands should be executed eagerly so that `sql("INSERT ...")` can trigger the
table insertion without a `.collect()`. We also need to eagerly execute non-root commands, because
many commands return `GenericInternalRow` and can't be put in a query plan directly, otherwise
the query engine may cast `GenericInternalRow` to `UnsafeRow` and fail
```